src/otx/algo/samplers/class_incremental_sampler.py

```diff
@@ -0,0 +1,181 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+"""Class incremental sampler for cls-incremental learning."""
+
+from __future__ import annotations
+
+import math
+
+import numpy as np
+import torch
+from torch.utils.data import Sampler
+
+from otx.core.data.dataset.base import OTXDataset
+from otx.core.utils.utils import get_idx_list_per_classes
+
+
+class ClassIncrementalSampler(Sampler):
+    """Sampler for Class-Incremental Task.
+
+    This sampler is a sampler that creates an effective batch
+    For default setting,
+    the square root of (number of old data/number of new data) is used as the ratio of old data
+    In effective mode,
+    the ratio of old and new data is used as 1:1
+
+    Args:
+        dataset (OTXDataset): A built-up dataset
+        batch_size (int): batch size of Sampling
+        efficient_mode (bool): Flag about using efficient mode
+        num_replicas (int, optional): Number of processes participating in
+            distributed training. By default, :attr:`world_size` is retrieved from the
+            current distributed group.
+        rank (int, optional): Rank of the current process within :attr:`num_replicas`.
+            By default, :attr:`rank` is retrieved from the current distributed
+            group.
+        drop_last (bool, optional): if ``True``, then the sampler will drop the
+            tail of the data to make it evenly divisible across the number of
+            replicas. If ``False``, the sampler will add extra indices to make
+            the data evenly divisible across the replicas. Default: ``False``.
+        n_repeats (Union[float, int, str], optional) : number of iterations for manual setting
+    """
+
+    def __init__(
+        self,
+        dataset: OTXDataset,
+        batch_size: int,
+        old_classes: list[str],
+        new_classes: list[str],
+        efficient_mode: bool = False,
+        num_replicas: int = 1,
+        rank: int = 0,
+        drop_last: bool = False,
+        n_repeats: int = 1,
+        generator: torch.Generator | None = None,
```

Comment:
Should user specify all of these values in config?

Reply:
Yes, that's right.
`data.config.train_subset.sampler.init_args` allows the user to set these values.